contracts/interchain-token-service/src/abi.rs

```diff
@@ -284,6 +269,18 @@ impl HubMessage {
                     message: Message::abi_decode(&decoded.message)?,
                 }
             }
+            MessageType::RegisterTokenMetadata => {
+                let RegisterTokenMetadata {
+                    tokenAddress,
+                    decimals,
+                    ..
+                } = RegisterTokenMetadata::abi_decode_params(payload, true)
+                    .map_err(Error::AbiDecodeFailed)?;
+                HubMessage::RegisterTokenMetadata(primitives::RegisterTokenMetadata {
+                    decimals,
+                    token_address: tokenAddress.to_vec().try_into().map_err(Error::NonEmpty)?,
```

Comment:
```suggestion
                    token_address: Vec::<u8>::from(tokenAddress).try_into().map_err(Error::NonEmpty)?,
```
You can use a conversion instead of needing to clone. Also applies to `LinkToken`